File: westcoast/src/main/java/frc/commands/BallHandlers/IntakeOut.java
/*----------------------------------------------------------------------------*/
/* Copyright (c) 2019 FIRST. All Rights Reserved.                             */
/* Open Source Software - may be modified and shared by FRC teams. The code   */
/* must be accompanied by the FIRST BSD license file in the root directory of */
/* the project.                                                               */
/*----------------------------------------------------------------------------*/

package frc.commands.BallHandlers;

import edu.wpi.first.wpilibj2.command.CommandBase;
import frc.subsystems.BallHandlingSubsystem;

public class IntakeOut extends CommandBase {
  private final BallHandlingSubsystem ball;

  public IntakeOut(
    BallHandlingSubsystem ballHandler
  ) {
    this.ball = ballHandler;
    addRequirements(ballHandler);
  }

  @Override
  public void initialize() {
  }

  @Override
  public void execute() {
    ball.runintakeMotor(-0.4);
  }

  @Override
  public void end(boolean interrupted) {
    ball.stopMotors();
  }

  @Override
  public boolean isFinished() {
    return false;
  }
}

File: westcoast/src/main/kotlin/frc/commands/magazine/IntakeOut.kt
package frc.commands.magazine

import edu.wpi.first.wpilibj2.command.CommandBase
import frc.subsystems.MagazineSubsystem

class IntakeOut(
    private val ball: MagazineSubsystem
) : CommandBase() {
  override fun initialize() {}
  override fun execute() {
    ball.runintakeMotor(-0.4)
  }

  override fun end(interrupted: Boolean) {
    ball.stopMotors()
  }

  override fun isFinished(): Boolean {
    return false
  }

  init {
    addRequirements(ball)
  }
}